### ISSUE TITLE
error for method ref of self type w/ Nothing parameter type

### ISSUE THREAD

**Issue by sgalles** [ceylon/ceylon-compiler#2361]
This code is useless but it should compile
```ceylon
shared interface Foo<Other> 
		given Other satisfies Foo<Other> {
	shared formal void bar(Other other);
}
shared void run(){
	Foo<Nothing> foo = nothing;
	value func = foo.bar; // error
}
```
```
method bar in interface Foo<Other> cannot be applied to given types;
  required: Foo
  found: java.lang.Object
  reason: actual argument java.lang.Object cannot be converted to Foo by method invocation conversion
```

**Comment by tombentley** [comment]
There's a whole class of related errors here, where a parameter gets javac-erased to its upper bound, but we just ignore that fact at method reference call sites. The best solution I can come up with is to figure out what type javac will erase the parameter to and use that. But that's not simple.

**Comment by tombentley** [comment]
Should now be fixed. Thanks @sgalles.

**Comment by tombentley** [comment]
Reopening this because it will be better to find another fix for this than to fix #2378

**Comment by FroMage** [comment]
Also much less `.src` files affected this time :)
